### PR TITLE
Pre-GA SDK for Azure LoadTesting

### PR DESCRIPTION
**Pull request by NiveditJain**
# Description
pre-ga SDK release for Azure LoadTesting

# Checklist 
- [x] Started PR 
- [x] Completed Directives 
- [x] Completed Generations
- [x] Custom Changes Done (including LRO Changes)
- [x] API View Done 
- [x] Documentations Updated 
- [x] Samples Updated 
- [x] Changelog Updated
- [x] Swagger Link Changed to main Repo
- [x] Added Tests and Make Necessary Updates 
- [x] Pipelines Fixed

### CODE REVIEW

**Comment by NiveditJain** [comment]
API View at: https://apiview.dev/Assemblies/Review/fc0755250d5e4c59b4b0a9e4d20dee53

File: sdk/loadtestservice/azure-developer-loadtesting/CHANGELOG.md
Comment (msyyc) [inline sdk/loadtestservice/azure-developer-loadtesting/CHANGELOG.md:3]: please add date before merge

File: sdk/loadtestservice/azure-developer-loadtesting/azure/developer/loadtesting/_patch.py
Comment (msyyc) [inline sdk/loadtestservice/azure-developer-loadtesting/azure/developer/loadtesting/_patch.py:37]: (1) Why are there some empty lines?
(2) Why don't you define it in swagger if they are used by some operations?

File: sdk/loadtestservice/azure-developer-loadtesting/CHANGELOG.md
Comment (NiveditJain) [reply sdk/loadtestservice/azure-developer-loadtesting/CHANGELOG.md]: added date here: https://github.com/Azure/azure-sdk-for-python/pull/27274/commits/c7cbbdaa9b137080e234800a70154c6bb80badf4

File: sdk/loadtestservice/azure-developer-loadtesting/setup.py
Comment (msyyc) [inline sdk/loadtestservice/azure-developer-loadtesting/setup.py:41]: please update the code with https://github.com/Azure/azure-sdk-for-python/wiki/Dataplane-Codegen-Quick-Start-with-tools#update and codegen will help to update `setup.py` automatically

File: sdk/loadtestservice/azure-developer-loadtesting/samples/get_test_run_metrics.py
Comment (msyyc) [inline sdk/loadtestservice/azure-developer-loadtesting/samples/get_test_run_metrics.py:58]: It is necessary to update and run test to check whether your code change works

File: sdk/loadtestservice/azure-developer-loadtesting/setup.py
Comment (NiveditJain) [reply sdk/loadtestservice/azure-developer-loadtesting/setup.py:41]: pushed code after running the scripts

File: sdk/loadtestservice/azure-developer-loadtesting/azure/developer/loadtesting/_patch.py
Comment (NiveditJain) [reply sdk/loadtestservice/azure-developer-loadtesting/azure/developer/loadtesting/_patch.py]: I am not sure, if I am able to understand you completely, can you please help me with it?

Comment (msyyc) [reply sdk/loadtestservice/azure-developer-loadtesting/azure/developer/loadtesting/_patch.py]: If you can define these enum value in swagger, autorest can generate it in `_enum.py` so that you don't need to write it manually. Could you provide context about these enum value?

**Comment by msyyc** [comment]
@NiveditJain I notice that you rename many operation. Could you run live test and commit your recording files?

File: sdk/loadtestservice/azure-developer-loadtesting/azure/developer/loadtesting/_patch.py
Comment (NiveditJain) [reply sdk/loadtestservice/azure-developer-loadtesting/azure/developer/loadtesting/_patch.py]: Hi @msyyc, 
There are 2 enums we are using.

First, TestFileValidationStatus, this is not in swagger its corresponding function is not also in swagger, this is for our customer wrapper/LRO which we are implementing, to check the validation status of loadtest script file (.JMX)

Second, this is in our swagger, not sure why it not being generated. 
image

File: sdk/loadtestservice/azure-developer-loadtesting/swagger/README.md
Comment (mikekistler) [inline sdk/loadtestservice/azure-developer-loadtesting/swagger/README.md:60]: Why are we doing all these transforms to the operationIds? Can't we use the operationIds as is?

Comment (msyyc) [reply sdk/loadtestservice/azure-developer-loadtesting/swagger/README.md:60]: Agree with mike. The more directives you use in README.md, the harder you maintain it in the future

File: sdk/loadtestservice/azure-developer-loadtesting/azure/developer/loadtesting/_patch.py
Comment (msyyc) [reply sdk/loadtestservice/azure-developer-loadtesting/azure/developer/loadtesting/_patch.py]: I believe it is related with your directives in README.md. Could you check it?

File: sdk/loadtestservice/azure-developer-loadtesting/azure/developer/loadtesting/_polling.py
Comment (Harshan01) [inline sdk/loadtestservice/azure-developer-loadtesting/azure/developer/loadtesting/_polling.py:160]: `return self._status in self._termination_statuses` looks cleaner IMO

File: sdk/loadtestservice/azure-developer-loadtesting/azure/developer/loadtesting/_polling.py
Comment (Harshan01) [inline sdk/loadtestservice/azure-developer-loadtesting/azure/developer/loadtesting/_polling.py:205]: Same. Direct return looks simpler

File: sdk/loadtestservice/azure-developer-loadtesting/README.md
Comment (ninallam) [inline sdk/loadtestservice/azure-developer-loadtesting/README.md:52]: Should we mention above that endpoint is the data plane URI?

File: sdk/loadtestservice/azure-developer-loadtesting/README.md
Comment (prativen) [inline sdk/loadtestservice/azure-developer-loadtesting/README.md:194]: Q: what does the result looked like if the poller times out at 3600 seconds

File: sdk/loadtestservice/azure-developer-loadtesting/CHANGELOG.md
Comment (prativen) [inline sdk/loadtestservice/azure-developer-loadtesting/CHANGELOG.md:10]: were we supposed to change app_components also to get_app_components since its dictionary and not an array

File: sdk/loadtestservice/azure-developer-loadtesting/README.md
Comment (prativen) [inline sdk/loadtestservice/azure-developer-loadtesting/README.md:62]: Q: why did we move away from 2 subclients as was earlier?

File: sdk/loadtestservice/azure-developer-loadtesting/samples/create_or_update_app_components.py
Comment (prativen) [inline sdk/loadtestservice/azure-developer-loadtesting/samples/create_or_update_app_components.py:20]: @ninallam should this be resource id of app component?

File: sdk/loadtestservice/azure-developer-loadtesting/CHANGELOG.md
Comment (NiveditJain) [reply sdk/loadtestservice/azure-developer-loadtesting/CHANGELOG.md:10]: Thanks for pointing this out! This should have been `get_app_components`, updating in ChangeLog

File: sdk/loadtestservice/azure-developer-loadtesting/README.md
Comment (NiveditJain) [reply sdk/loadtestservice/azure-developer-loadtesting/README.md:62]: As per the API View comment, 

```Should the administration operations be a separate administration client?

If so, the two clients should flatten away these operation groups.


If we believe customers will use these together in a single client, I would recommend dropping 'load_test' from the operation group names, as it's already in the client name, and makes the names very long.``` by @annatisch 

The idea was that in C# we agreed to make 2 clients, so trying to be consistent.

File: sdk/loadtestservice/azure-developer-loadtesting/README.md
Comment (NiveditJain) [reply sdk/loadtestservice/azure-developer-loadtesting/README.md:194]: @prativen `LROPoller.result()` works in a way that it waits for a maximum of given number of seconds, and then returns the last response received. So, if poller time's out at 3600 seconds then it would return the result of last request before the timeout.

File: sdk/loadtestservice/azure-developer-loadtesting/CHANGELOG.md
Comment (NiveditJain) [reply sdk/loadtestservice/azure-developer-loadtesting/CHANGELOG.md]: Yes, updated!

File: sdk/loadtestservice/azure-developer-loadtesting/CHANGELOG.md
Comment (NiveditJain) [reply sdk/loadtestservice/azure-developer-loadtesting/CHANGELOG.md]: I am unable to understand this comment fully as I am unable to find these lines in the latest code, can you please help me?

File: sdk/loadtestservice/azure-developer-loadtesting/azure/developer/loadtesting/aio/operations/_operations.py
Comment (NiveditJain) [reply sdk/loadtestservice/azure-developer-loadtesting/azure/developer/loadtesting/aio/operations/_operations.py]: as per the latest version this line is, 
```
def build_load_test_administration_create_or_update_app_components_request(test_id: str, **kwargs: Any) -> HttpRequest:
```

File: sdk/loadtestservice/azure-developer-loadtesting/azure/developer/loadtesting/aio/operations/_operations.py
Comment (NiveditJain) [reply sdk/loadtestservice/azure-developer-loadtesting/azure/developer/loadtesting/aio/operations/_operations.py]: as per the latest code this line is, `get_test_file`

File: sdk/loadtestservice/azure-developer-loadtesting/samples/create_or_update_app_components.py
Comment (NiveditJain) [reply sdk/loadtestservice/azure-developer-loadtesting/samples/create_or_update_app_components.py]: Yes, this is patch API. For `get` we also have a different API but it does `overwrites`

File: sdk/loadtestservice/azure-developer-loadtesting/samples/create_or_update_test_run.py
Comment (NiveditJain) [reply sdk/loadtestservice/azure-developer-loadtesting/samples/create_or_update_test_run.py]: @prativen I guess you are not checking the latest commits, as per the latest commits pushed to gh, this line is, 

```
testRunPoller = client.begin_create_or_update_test_run(
    TEST_RUN_ID,
    {
        "testId": TEST_ID,
        "displayName": "My New Load Test Run",
    },
    poll_for_test_run_status=True
)
```

File: sdk/loadtestservice/azure-developer-loadtesting/samples/create_or_update_app_components.py
Comment (NiveditJain) [reply sdk/loadtestservice/azure-developer-loadtesting/samples/create_or_update_app_components.py]: Yes we are not doing so, this was pushed by mistake while testing on df, removed them.

File: sdk/loadtestservice/azure-developer-loadtesting/samples/hero.py
Comment (NiveditJain) [reply sdk/loadtestservice/azure-developer-loadtesting/samples/hero.py]: We have changed the approach for LROPoller, this is no longer valid. Thanks!

File: sdk/loadtestservice/azure-developer-loadtesting/samples/hero.py
Comment (NiveditJain) [reply sdk/loadtestservice/azure-developer-loadtesting/samples/hero.py]: added comments here!

File: sdk/loadtestservice/azure-developer-loadtesting/azure/developer/loadtesting/_polling.py
Comment (NiveditJain) [reply sdk/loadtestservice/azure-developer-loadtesting/azure/developer/loadtesting/_polling.py:31]: I think we can do the refactoring here. Let me do that.

**Comment by NiveditJain** [comment]
> @NiveditJain I notice that you rename many operation. Could you run live test and commit your recording files?

@msyyc I have added tests!

File: sdk/loadtestservice/azure-developer-loadtesting/azure/developer/loadtesting/_polling.py
Comment (NiveditJain) [reply sdk/loadtestservice/azure-developer-loadtesting/azure/developer/loadtesting/_polling.py:31]: Resolved in the latest commit.

File: sdk/loadtestservice/azure-developer-loadtesting/azure/developer/loadtesting/_patch.py
Comment (msyyc) [inline sdk/loadtestservice/azure-developer-loadtesting/azure/developer/loadtesting/_patch.py:33]: @iscai-msft 
@NiveditJain patch the operation group into client which is not done before. Although it is a little weird, the code works. What do you think about it?

File: sdk/loadtestservice/azure-developer-loadtesting/azure/developer/loadtesting/_polling.py
Comment (msyyc) [inline sdk/loadtestservice/azure-developer-loadtesting/azure/developer/loadtesting/_polling.py:21]: Could you move the code into `_patch.py`? Otherwise it may be overrided or cleared by codegen.

Comment (NiveditJain) [reply sdk/loadtestservice/azure-developer-loadtesting/azure/developer/loadtesting/_polling.py]: Done thanks for reminding me. I referred `_polling.py` from Azure Keyvault, just want to bring to notice that it might not be generated code. Changed pushed in latest commit.

File: sdk/loadtestservice/azure-developer-loadtesting/samples/create_or_update_app_components.py
Comment (NiveditJain) [reply sdk/loadtestservice/azure-developer-loadtesting/samples/create_or_update_app_components.py]: @prativen can I mark this as resolved?

File: sdk/loadtestservice/azure-developer-loadtesting/azure/developer/loadtesting/aio/operations/_operations.py
Comment (NiveditJain) [reply sdk/loadtestservice/azure-developer-loadtesting/azure/developer/loadtesting/aio/operations/_operations.py]: @prativen can I mark this as resolved?

File: sdk/loadtestservice/azure-developer-loadtesting/azure/developer/loadtesting/aio/operations/_operations.py
Comment (NiveditJain) [reply sdk/loadtestservice/azure-developer-loadtesting/azure/developer/loadtesting/aio/operations/_operations.py]: @prativen can I mark this as resolved?

**Review by ninallam — APPROVED** [review]

**Comment by msyyc** [comment]
Hi @NiveditJain The PR overall looks good. I notice that there is empty file `_polling.py` which I think shall be removed.

**Comment by NiveditJain** [comment]
> Hi @NiveditJain The PR overall looks good. I notice that there is empty file `_polling.py` which I think shall be removed.

Hi @msyyc I have removed the file, can you please mark this PR as approved if there are no other comments?

File: sdk/loadtestservice/azure-developer-loadtesting/azure/developer/loadtesting/_patch.py
Comment (msyyc) [reply sdk/loadtestservice/azure-developer-loadtesting/azure/developer/loadtesting/_patch.py:33]: @iscai-msft The example indeed have better structure, while @NiveditJain think they need to copy all operations code into the _patch.py which is very inconvenient and makes the _patch.py swollen. From usage experience, I think his method make sense.

Comment (iscai-msft) [reply sdk/loadtestservice/azure-developer-loadtesting/azure/developer/loadtesting/_patch.py:33]: We don't want to expose those operation groups in the operations folder though. Even though it's a lot grosser code, we don't want an `operations` folder, i.e. `azure.loadtesting.operations.LoadTestAdministrationOperations` etc. The only way to get rid of the folder would be to generate with all operations on a single mixin operations class, then give each exposed client access to the mixin operation (but don't inherit, because this will expose extra operations on each client), divide up the operations between the clients, and expose them to users that way

**Review by iscai-msft — APPROVED** [review]

File: sdk/loadtestservice/azure-developer-loadtesting/README.md
Comment (iscai-msft) [inline sdk/loadtestservice/azure-developer-loadtesting/README.md:70]: nit: `.administration`

File: sdk/loadtestservice/azure-developer-loadtesting/CHANGELOG.md
Comment (iscai-msft) [inline sdk/loadtestservice/azure-developer-loadtesting/CHANGELOG.md:20]: you should say which operation group these are on, for example is this `LoadTestRunClient.administration.get_app_components`

File: sdk/loadtestservice/azure-developer-loadtesting/azure/developer/loadtesting/aio/operations/_operations.py
Comment (iscai-msft) [inline sdk/loadtestservice/azure-developer-loadtesting/azure/developer/loadtesting/aio/operations/_operations.py:1442]: can you run the samples? You have patched this operation as `upload_test_file`, but you sample is running `begin_upload_test_file`. The first is a regular operation, and the sample is an LRO operation. Which one is it supposed to be?

File: sdk/loadtestservice/azure-developer-loadtesting/azure/developer/loadtesting/aio/operations/_operations.py
Comment (iscai-msft) [inline sdk/loadtestservice/azure-developer-loadtesting/azure/developer/loadtesting/aio/operations/_operations.py:2238]: i do also have a question about all of the patched operations. Why are so many being patched, are they not part of the swagger?

**Review by iscai-msft — APPROVED** [review]
merging for now, going to have a separate PR to fix the double operations we're running into

**Pull request merged**
